src/net/server/channel/handlers/FamilyUseHandler.java

```diff
@@ -21,82 +21,117 @@
 */
 package net.server.channel.handlers;
 
-import constants.ServerConstants;
 import client.MapleCharacter;
 import client.MapleClient;
+import client.MapleFamilyEntitlement;
+import client.MapleFamilyEntry;
+import constants.ServerConstants;
 import net.AbstractMaplePacketHandler;
-import net.opcodes.SendOpcode;
+import net.server.coordinator.MapleInviteCoordinator;
+import net.server.coordinator.MapleInviteCoordinator.InviteType;
+import server.maps.FieldLimit;
+import server.maps.MapleMap;
+import tools.MaplePacketCreator;
 import tools.data.input.SeekableLittleEndianAccessor;
-import tools.data.output.MaplePacketLittleEndianWriter;
 
 /**
  *
  * @author Moogra
+ * @author Ubaware
  */
 public final class FamilyUseHandler extends AbstractMaplePacketHandler {
     @Override
     public final void handlePacket(SeekableLittleEndianAccessor slea, MapleClient c) {
-    	if (!ServerConstants.USE_FAMILY_SYSTEM){
-    		return;
-    	}
-        int[] repCost = {3, 5, 7, 8, 10, 12, 15, 20, 25, 40, 50};
-        final int type = slea.readInt();
+        if(!ServerConstants.USE_FAMILY_SYSTEM) {
+            return;
+        }
+        MapleFamilyEntitlement type = MapleFamilyEntitlement.values()[slea.readInt()];
+        int cost = type.getRepCost();
+        MapleFamilyEntry entry = c.getPlayer().getFamilyEntry();
+        if(entry.getReputation() < cost || entry.isEntitlementUsed(type)) {
+            return; // shouldn't even be able to request it
+        }
+        c.announce(MaplePacketCreator.getFamilyInfo(entry));
         MapleCharacter victim;
-        if (type == 0 || type == 1) {
+        if(type == MapleFamilyEntitlement.FAMILY_REUINION || type == MapleFamilyEntitlement.SUMMON_FAMILY) {
             victim = c.getChannelServer().getPlayerStorage().getCharacterByName(slea.readMapleAsciiString());
-            if (victim != null) {
-                if (type == 0) {
-                    c.getPlayer().changeMap(victim.getMap(), victim.getMap().getPortal(0));
-                } else {
-                    victim.changeMap(c.getPlayer().getMap(), c.getPlayer().getMap().getPortal(0));
+            if(victim != null && victim != c.getPlayer()) {
+                MapleMap targetMap = victim.getMap();
```

Comment:
Should check if players are in the same family. Send family message 67 if not

Reply:
Thanks, I completely overlooked that. Done.